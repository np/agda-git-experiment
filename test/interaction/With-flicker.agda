module With-flicker where

data ℕ : Set where
  zero : ℕ
  suc  : ℕ → ℕ

_+_ : ℕ → ℕ → ℕ
zero  + n = n
suc m + n = suc (m + n)

fib : ℕ → ℕ
fib zero          = zero
fib (suc zero)    = suc zero
fib (suc (suc n)) = fib n + fib (suc n)

thirteen = suc (suc (suc (suc (suc (suc (suc (suc (suc (suc (suc (suc (suc zero))))))))))))

postulate
  P : ℕ → Set
  n : ℕ

-- The interactive highlighting used to flicker a lot when the
-- following definition was checked.

Foo : P (fib (thirteen + n)) → Set₁
Foo p with zero
Foo p | _ = Set

-- TODO: Make sure that interactive highlighting output is included in
-- With-flicker.out.
