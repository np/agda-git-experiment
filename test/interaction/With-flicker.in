IOTCM currentFile Interactive Direct (cmd_load currentFile [])
