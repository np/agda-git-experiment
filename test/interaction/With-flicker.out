Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking With-flicker (With-flicker.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished With-flicker.\n" t)
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "" nil)
((last . 1) . (agda2-goals-action '()))
Agda2> 